libr/asm/p/asm_x86_nz.c

```diff
@@ -517,6 +517,134 @@ static int opcall(RAsm *a, ut8 *data, const Opcode op) {
 	return l;
 }
 
+static int opcmov(RAsm *a, ut8 *data, const Opcode op) {
+	int l = 0;
+	int mod_byte = 0;
+	int offset = 0;
+
+	if (op.operands[0].type & OT_MEMORY ||
+	    op.operands[1].type & OT_CONSTANT) {
+		return -1;
+	}
+
+	data[l++] = 0x0f;
+	if (!strcmp (op.mnemonic, "cmovo")) {
```

Comment:
With a switch char by char I wouldn't be able to group instructions that have the same opcode.
I'd prefer checking the suffixes only